src/main/java/io/strimzi/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -134,16 +145,14 @@ protected void initConsumer(boolean shouldAttachBatchHandler, Properties config)
         KafkaConfig consumerConfig = this.bridgeConfigProperties.getKafkaConfig();
         Properties props = new Properties();
         props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, consumerConfig.getBootstrapServers());
-        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, consumerConfig.getConsumerConfig().getKeyDeserializer());
-        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, consumerConfig.getConsumerConfig().getValueDeserializer());
         props.put(ConsumerConfig.GROUP_ID_CONFIG, this.groupId);
         props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, consumerConfig.getConsumerConfig().isEnableAutoCommit());
         props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, consumerConfig.getConsumerConfig().getAutoOffsetReset());
 
         if (config != null)
             props.putAll(config);
 
-        this.consumer = KafkaConsumer.create(this.vertx, props);
+        this.consumer = KafkaConsumer.create(this.vertx, props, keyDeserializer, valueDeserializer);
```

Comment:
Can't we put `keyDeserializer` and `valueDeserializer` into `props`?

Reply:
They were in the props but you know that in that case, we have to specify the class name; here the deserializers are passed from outside based on the requested format (json or binary).